src/main/java/com/library/service/implementation/BookServiceImpl.java

```diff
@@ -0,0 +1,43 @@
+package com.library.service.implementation;
+
+import com.library.model.dto.BookDto;
+import com.library.model.entity.Book;
+import com.library.repository.BookRepository;
+import com.library.service.BookService;
+import lombok.RequiredArgsConstructor;
+import org.modelmapper.ModelMapper;
+import org.springframework.stereotype.Service;
+
+import jakarta.persistence.EntityNotFoundException;
+
+@Service
+@RequiredArgsConstructor
+public class BookServiceImpl implements BookService {
+
+    private final BookRepository bookRepository;
+
+    private final ModelMapper modelMapper;
+
+
+    @Override
+    public BookDto findById(Long id) {
+        return modelMapper.map(bookRepository.findById(id).orElseThrow(EntityNotFoundException::new), BookDto.class);
```

Comment:
Please look at AuthotServiseImpl.
Not similar return.

Reply:
What do you mean? Method from AuthorServiceImpl class return AuthorDto and BookServiceImpl - BookDto...

Reply:
will discuss on call.